### DISCUSSION TITLE
A right way to pass settings to plugin

### DISCUSSION THREAD

**Discussion by Cyrille37** [alpinejs/alpine#4176] [1. Help] [answered]
Hi all :-)

What would be the best way to pass settings to plugin ?

I'm proposing a setting for the plugin [markmead/alpinejs-component](https://github.com/markmead/alpinejs-component/issues/27). I would like to know if my proposed solution follow the right way.

```javascript
import Alpine from 'alpinejs'
import theNicePlugin from 'alpinejs-niceplugin'
// Store the setting in Alpine :
Alpine.store('alpinejs-niceplugin.settings', {foo: 1,bar:"hello"} );
Alpine.plugin(theNicePlugin)
```

It works in ESM environnement but I don't know how it could be used with CDN as there is no initialization time but only scripts loading like:

```html
<script defer src="https://unpkg.com/alpinejs-component@latest/dist/component.min.js"></script>
<script defer src="https://unpkg.com/alpinejs@3.x.x/dist/cdn.min.js"></script>
```

Thanks for your comment :-)

Regards & cheers

**Comment by ekwoka** [comment, marked as answer]
why not have it be passed to the plugin calling it as a function?

```js
Alpine.plugin(RIAS({ shopify: true, autoSize: true }))
```

for instance

or if its a component, why not have it passed to the component when it's instantiated?

for cdn use, you can use attributes on the script element itself with [document.currentScript](https://developer.mozilla.org/en-US/docs/Web/API/Document/currentScript) to access the script and get the attributes.

**Reply by Cyrille37** [reply]
Thanks for your lights :-)
